fee-abstraction/cmd/hook.go

```diff
@@ -0,0 +1,65 @@
+package cmd
+
+import (
+	"context"
+
+	"github.com/blang/semver/v4"
+	"github.com/ignite/cli/v28/ignite/pkg/cliui"
+	"github.com/ignite/cli/v28/ignite/pkg/gomodulepath"
+	"github.com/ignite/cli/v28/ignite/pkg/placeholder"
+	"github.com/ignite/cli/v28/ignite/services/chain"
+	"github.com/ignite/cli/v28/ignite/services/plugin"
+
+	"github.com/ignite/apps/fee-abstraction/services/scaffolder"
+)
+
+// ExecuteScaffoldChainHook executes the scaffold chain hook.
+func ExecuteScaffoldChainHook(ctx context.Context, h *plugin.ExecutedHook, api plugin.ClientAPI) error {
+	var (
+		flags           = plugin.Flags(h.Hook.Flags)
+		feeAbsModule, _ = flags.GetBool(flagFeeAbsModule)
+		name            = h.ExecutedCommand.Args[0]
+	)
+	if !feeAbsModule {
+		return nil
+	}
+
+	session := cliui.New(cliui.StartSpinnerWithText(statusScaffolding))
+	defer session.End()
+
+	pathInfo, err := gomodulepath.Parse(name)
+	if err != nil {
+		return err
+	}
+
+	version := getVersion(flags)
+	semVersion, err := semver.Parse(version)
+	if err != nil {
+		return err
+	}
+
+	c, err := newChain(pathInfo.Root, flags, chain.WithOutputer(session), chain.CollectEvents(session.EventBus()))
```

Comment:
Let's add an SDK version check like done in rollkit. We eventually need to do this in all apps. This specific scaffolding logic works only with 0.50.

Later we can switch on different sdk version and have different templates

Reply:
we alredy have here https://github.com/ignite/apps/pull/130/files#diff-6906af81ecebb9a185c7aa645f3ca0aee53387340299344e8ab405b208604734R40-R45

Reply:
Yes but don't want it to be greather than v0.50 either.